src/components/ThumbnailImage.js

```diff
@@ -33,73 +31,71 @@ const defaultProps = {
-        // For portrait images, we need to adjust the width of the image to keep the aspect ratio and vice-versa.
-        if (thumbnailScreenHeight > thumbnailScreenWidth) {
-            thumbnailScreenWidth = Math.round(thumbnailScreenHeight * (1 / aspectRatio));
-        } else {
-            thumbnailScreenHeight = Math.round(thumbnailScreenWidth * aspectRatio);
-        }
-        return {thumbnailWidth: Math.max(40, thumbnailScreenWidth), thumbnailHeight: Math.max(40, thumbnailScreenHeight)};
+    // If thumbnail height is greater than its width, then the image is portrait otherwise landscape.
+    // For portrait images, we need to adjust the width of the image to keep the aspect ratio and vice-versa.
+    if (thumbnailScreenHeight > thumbnailScreenWidth) {
+        thumbnailScreenWidth = Math.round(thumbnailScreenHeight * (1 / aspectRatio));
+    } else {
+        thumbnailScreenHeight = Math.round(thumbnailScreenWidth * aspectRatio);
     }
+    return {thumbnailWidth: Math.max(40, thumbnailScreenWidth), thumbnailHeight: Math.max(40, thumbnailScreenHeight)};
+}
+
+function ThumbnailImage(props) {
+    const {windowHeight} = useWindowDimensions();
+    const initialDimensions = calculateThumbnailImageSize(props.imageWidth, props.imageHeight, windowHeight);
+    const [imageWidth, setImageWidth] = useState(initialDimensions.thumbnailWidth);
+    const [imageHeight, setImageHeight] = useState(initialDimensions.thumbnailHeight);
 
     /**
      * Update the state with the computed thumbnail sizes.
      *
      * @param {{ width: number, height: number }} Params - width and height of the original image.
      */
-    updateImageSize({width, height}) {
-        const {thumbnailWidth, thumbnailHeight} = this.calculateThumbnailImageSize(width, height);
-        this.setState({thumbnailWidth, thumbnailHeight});
-    }
 
-    render() {
-        return (
-            <View style={[this.props.style, styles.overflowHidden]}>
-                <View style={[StyleUtils.getWidthAndHeightStyle(this.state.thumbnailWidth, this.state.thumbnailHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
-                    <ImageWithSizeCalculation
-                        url={this.props.previewSourceURL}
-                        onMeasure={this.updateImageSize}
-                        isAuthTokenRequired={this.props.isAuthTokenRequired}
-                    />
-                </View>
+    const updateImageSize = useCallback(
+        ({width, height}) => {
+            const {thumbnailWidth, thumbnailHeight} = calculateThumbnailImageSize(width, height, windowHeight);
+            setImageWidth(thumbnailWidth);
+            setImageHeight(thumbnailHeight);
+        },
+        [windowHeight],
+    );
+    return (
+        <View style={[props.style, styles.overflowHidden]}>
+            <View style={[StyleUtils.getWidthAndHeightStyle(imageWidth || 200, imageHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
```

Comment:
Why we have this fallback `imageWidth || 200`? It was not there before. Your commit indicates it's about some undefined state case, can you elaborate on that?

Reply:
This was included to resolve the following error that was occurring with `imageWidth`:

```
Argument of type 'number | undefined' is not assignable to parameter of type 'number'. Type 'undefined' is not assignable to type 'number'
```

Reply:
This is a bug on `main` as well right? Let's either fix the root cause, I assume this is coming from `calculateThumbnailImageSize` where it returns an empty object if width or height is not truthy. Or work on this separately in a new issue.

Reply:
Thansk for that! I've updated the `calculateThumbnailImageSize` function to return a value instead of being empty.

Reply:
@kadiealexander Thanks for the update! Can you please include the steps to reproduce the bug? I don't see any errors from my end

Reply:
I believe it was just a lint error, not a bug. The linter was complaining that `imageWidth` might not be set.

Reply:
Ah okay. I think we can move on with a default 200x200. BTW, is there any real case where the width or height is not set? I thought this would be the case with svg files but those does not have a thumbnail preview

Reply:
I don't think so? We upload SVGs as files, which don't have a thumbnail. I feel like all other image types should have a fixed height and width.

Reply:
So there is no way that width and height could be undefined in `calculateThumbnailImageSize()`, right?
If so, can we remove that?

Reply:
@MonilBhavsar we're not bothered about the lint error in this instance?

Reply:
I think linter error was there because we were returning an empty object, and then trying to access the offset.
We were returning empty object if either of width or height is undefined. I feel like width and height of image can never be undefined and if we can skip the check completely - 
```js
if (!width || !height) {
    return {thumbnailWidth: 200, thumbnailHeight: 200};
}
```
But, NAB 👍